### PR TITLE
feat: Update dependencies, adapt tooling and regenerate

### PR DESCRIPTION
**Pull request by jskeet**
(One commit at a time...)

### CODE REVIEW

File: tools/Google.Events.Tools.CodeGenerator/Program.cs [hunk 1]
```diff
@@ -99,9 +99,18 @@ void GenerateCloudEventTypes()
                         writer.WriteLine("    {");
                         foreach (var type in group.Select(ce => ce.Type))
                         {
-                            var suffix = type.Split('.').Last();
+                            // Firestore and Datastore have separate "withAuthContext" variants; in those cases we want to
+                            // use the previous segment as well. Hard-coding this isn't pleasant, but it'll do until we see
+                            // anythign else we need to do.
```
Comment (amanda-tarafa) [inline tools/Google.Events.Tools.CodeGenerator/Program.cs:104]: tiny typo: anythign => anything

File: tools/Google.Events.Tools.CodeGenerator/Program.cs [hunk 2]
```diff
@@ -99,9 +99,18 @@ void GenerateCloudEventTypes()
...
+                            var suffix = type
+                                .Replace(".withAuthContext", "WithAuthContext")
+                                .Split('.')
+                                .Last();
+                            // "Undo" the hack before to get back to the relevant part of the type.
+                            // (We can't just use type.Split('.').Last(), as that would just give us "withAuthContext".)
+                            var description = suffix.Replace("WithAuthContext", ".withAuthContext");
```
Comment (amanda-tarafa) [inline tools/Google.Events.Tools.CodeGenerator/Program.cs:111]: description is now `foo.withAuthContext`, just making sure that's intended.

**Review by amanda-tarafa — APPROVED** [review]

File: tools/Google.Events.Tools.CodeGenerator/Program.cs (hunk 1 above)
Comment (jskeet) [reply tools/Google.Events.Tools.CodeGenerator/Program.cs:104]: Whoops, thanks - will fix in the morning.

File: tools/Google.Events.Tools.CodeGenerator/Program.cs (hunk 2 above)
Comment (jskeet) [reply tools/Google.Events.Tools.CodeGenerator/Program.cs:111]: Yup, that's correct. (If you have a look in the context of the other constants, I think that's probably the best option.)

**Pull request merged**